### DISCUSSION TITLE
Specific "First citizens" RDBMS

### DISCUSSION THREAD

**Discussion by manticore-projects** [JSQLParser/JSqlParser#1247] [Ideas]
Greetings.

Responding to a lot of issues, I feel like there is a lot of expectation for support of rather exotic RDBMS and syntax.
JSQLParser does not state, what exactly an user should expect and there is no clear objective.

Maybe we would like to:

1) Define a major goal of support and compatibility (e. g. SQL99 standard + Oracle 11g + MS SQL Server + PostgreSQL)
I would call these "First Citizen".

2) Define a second tier compatibility for features, that are nice to have and do not collide with item 1 -- maybe H2 and MySQL

3) Define anything out of scope, which would collide with items 1) or 2) or is simply too exotic/unusual/outdated (e. g. Sybase)

**Comment by wumpz** [comment]
This should be done using validators. Look for instance into PostgresqlVersion.java.

**Reply by manticore-projects** [reply]
This is not what I mean.
I was talking about "managing expectations" of the end users. We get a lot of issues reported for rather exotic RDBMS and right now there is no policy or scope definition.
I am talking about pointing out such a scope on the website, so end-user know what should be expected and supported and what not.

**Reply by wumpz** [reply]
Understood, that's why your interesting idea with those rr diagramms. So this is not only programming more of documenting stuff, right?

**Reply by wumpz** [reply]
Sorry about my correspondence time. Here is so much to do.

**Reply by manticore-projects** [reply]
> Understood, that's why your interesting idea with those rr diagramms. So this is not only programming more of documenting stuff, right?

Yes, documentation and expectation management.
It seems, like you and me, we are both `Oracle guys` and for use Oracle presents the normal view on SQL. Although we know that H2 and Postgres exist we have a certain bias leaning towards Oracle.

A MySQL guy would see the support of all the exotic MySQL features as essential and can't even imagine, that JSQLParser would not immediately support all of that. (And Oracle or MySQL are just names here, please replace it with your own preferred brand or product.)

So I would like to document clearly, what should be expected (e. g. all Oracle features up to 12.2) and thus would be a bug (which we care about immediately) when not working.

Or, what would be nice to have, but due to resource constraints would be supported only when sponsored.

Or, what is just too exotic and so out of scope unless someone sends a PR.

The Railroad diagrams can help with that understanding, although they would provide a second level of detailed documentation. Maybe we should provide  a higher level of scope before that.
